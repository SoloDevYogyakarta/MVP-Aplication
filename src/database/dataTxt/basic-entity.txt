{"id":539,"name":"Tasty Plastic Car","status":2,"condition":0,"shortdesc":"Possimus alienus solium ciminatio tergo convoco utroque tunc decerno accedo. Tego ara aperte thymum compello reiciendis collum volubilis asperiores. Tenetur quod conqueror atrocitas advenio ullus tenuis aqua.","main_stock":1,"reserve_stock":2,"user_id":"Hk3YcUFZgDO1e9H3u9B16","updatedAt":"2023-10-26T06:54:51.972Z","createdAt":"2023-10-26T06:54:51.972Z","public_id":"d4Z1Evq2z1k0xSbN02Dsi"}