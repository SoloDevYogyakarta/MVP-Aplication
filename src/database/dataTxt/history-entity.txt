{"id":11,"public_id":"MLf4_sOkio9GAfkIwj4tk","name":"Fantastic Steel Bacon","desc":"New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart","user_id":"Hk3YcUFZgDO1e9H3u9B16","updatedAt":"2023-10-26T06:54:52.587Z","createdAt":"2023-10-26T06:54:52.587Z"}