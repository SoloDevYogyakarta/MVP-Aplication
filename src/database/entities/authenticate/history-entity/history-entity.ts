import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserHistoryEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  address: string;
  user_id: number;
}

export type UserHistoryCreationAttribute = Optional<UserHistoryEntity, 'id'>;
export interface UserHistoryInstance
  extends Model<UserHistoryCreationAttribute, UserHistoryEntity>,
    UserHistoryEntity {}
export const userHistoryEntity = sequelize.define<UserHistoryInstance>(
  'HISTORY',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    address: DataTypes.STRING,
    user_id: DataTypes.INTEGER,
  } as ModelAttributes<UserHistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'AUTHENTICATE',
    timestamps: false,
  },
);
