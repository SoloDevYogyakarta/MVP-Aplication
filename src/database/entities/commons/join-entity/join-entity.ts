import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class JoinEntity extends BaseEntity {
  source_id: string;
  foreign_id: string;
}

export type JoinCreationAttribute = Optional<JoinEntity, 'id'>;
export interface JoinInstance
  extends Model<JoinCreationAttribute, JoinEntity>,
    JoinEntity {}
export const joinEntity = sequelize.define<JoinInstance>(
  'JOIN',
  {
    public_id: DataTypes.STRING,
    source_id: DataTypes.STRING,
    foreign_id: DataTypes.STRING,
  } as ModelAttributes<JoinInstance>,
  {
    tableName: 'JOIN',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
