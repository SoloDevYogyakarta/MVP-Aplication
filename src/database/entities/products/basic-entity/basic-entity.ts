import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ProductBasicEntity extends BaseEntity {
  name?: string;
  status?: number;
  condition?: number;
  shortdesc?: string;
  main_stock?: number;
  reserve_stock?: number;
  user_id?: string;
}

export type ProductBasicCreationAttribute = Optional<ProductBasicEntity, 'id'>;
export interface ProductBasicInstance
  extends Model<ProductBasicCreationAttribute, ProductBasicEntity>,
    ProductBasicEntity {}
export const productBasicEntity = sequelize.define<ProductBasicInstance>(
  'BASIC',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    status: DataTypes.INTEGER,
    condition: DataTypes.INTEGER,
    shortdesc: DataTypes.STRING,
    main_stock: DataTypes.INTEGER,
    reserve_stock: DataTypes.INTEGER,
    user_id: DataTypes.STRING,
  } as ModelAttributes<ProductBasicInstance>,
  {
    tableName: 'BASIC',
    schema: 'PRODUCTS',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
