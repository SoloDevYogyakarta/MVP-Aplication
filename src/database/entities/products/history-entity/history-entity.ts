import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  type: string;
  date: Date;
  product_id: string;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    type: DataTypes.STRING,
    date: DataTypes.DATE,
    product_id: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
