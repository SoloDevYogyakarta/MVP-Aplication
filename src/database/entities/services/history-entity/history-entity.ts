import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  name: string;
  desc: string;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    desc: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'SERVICES',
  },
);
