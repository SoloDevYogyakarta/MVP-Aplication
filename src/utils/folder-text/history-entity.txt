{"id":16,"name":"Handmade Wooden Mouse","title":"Bronze","desc":"The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design","price":69653,"order_id":8,"updatedAt":"2023-11-07T07:15:16.412Z","createdAt":"2023-11-07T07:15:16.412Z"}