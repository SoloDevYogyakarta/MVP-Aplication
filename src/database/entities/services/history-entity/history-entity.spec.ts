import { faker } from '@faker-js/faker';
import { ModelCtor } from 'sequelize';
import { createpath } from '../../../../utils/system/system';
import { getField } from '../../../../utils/get-field/get-field';
import { joinEntity } from '../../commons/join-entity/join-entity';
import { productBasicEntity } from '../../products/basic-entity/basic-entity';
import { historyEntity, HistoryInstance } from './history-entity';
import { nanoid } from 'nanoid';

describe('HistoryEntity', () => {
  let public_id!: string;
  let entity: ModelCtor<HistoryInstance>;

  beforeEach(() => {
    entity = historyEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  it('create', async () => {
    const user_id = getField('user-http-entity').public_id;
    const products = await productBasicEntity.findAll({
      offset: faker.number.int({ min: 1, max: 10 }),
      limit: 2,
    });
    const history = await entity.create({
      public_id: nanoid(),
      name: faker.commerce.productName(),
      desc: faker.commerce.productDescription(),
      user_id,
    });
    history.save();
    for (const product of products) {
      const join = await joinEntity.create({
        source_id: history.public_id,
        foreign_id: product.public_id,
      });
      join.save();
    }

    const secondhistory = await entity.create({
      public_id: nanoid(),
      name: faker.commerce.productName(),
      desc: faker.commerce.productDescription(),
      user_id,
    });
    history.save();
    for (const product of products) {
      const join = await joinEntity.create({
        source_id: secondhistory.public_id,
        foreign_id: product.public_id,
      });
      join.save();
    }

    createpath(
      '../../database/dataTxt/second-history-entity.txt',
      secondhistory,
    );
    createpath('../../database/dataTxt/history-entity.txt', history);
    expect(history.user_id).toEqual(user_id);
  });

  try {
    public_id = getField('history-entity').public_id;
  } catch (err) {}

  if (public_id) {
    it('findOne', async () => {
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  }
});
