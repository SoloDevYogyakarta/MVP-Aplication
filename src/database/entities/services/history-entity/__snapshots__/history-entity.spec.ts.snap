// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryEntity render correctly 1`] = `[Function]`;
